src/common/com/intellij/plugins/haxe/util/HaxeResolveUtil.java

```diff
@@ -526,8 +526,13 @@ public static HaxeClass tryResolveClassByQName(@Nullable PsiElement type) {
       return null;
     }
 
+    // can be accessed without full QName
+    HaxeClass result = findClassByQName(type.getText(), type.getContext());
```

Comment:
Well, no time like the present.  8)

Since rewriting the resolver is high on my "want to do" list, extra unit tests around it will help to ensure I don't re-break it when I finally get around to doing so.  I would appreciate that test, if it's not much trouble.